drive redirection not working on RHEL 9

### xrdp version

0.9.25

### Detailed xrdp version, build options

~~~
$ xrdp --version
xrdp 0.9.25
  A Remote Desktop Protocol Server.
  Copyright (C) 2004-2020 Jay Sorg, Neutrino Labs, and all contributors.
  See https://github.com/neutrinolabs/xrdp for more information.

  Configure options:
      --build=x86_64-redhat-linux-gnu
      --host=x86_64-redhat-linux-gnu
      --program-prefix=
      --disable-dependency-tracking
      --prefix=/usr
      --exec-prefix=/usr
      --bindir=/usr/bin
      --sbindir=/usr/sbin
      --sysconfdir=/etc
      --datadir=/usr/share
      --includedir=/usr/include
      --libdir=/usr/lib64
      --libexecdir=/usr/libexec
      --localstatedir=/var
      --sharedstatedir=/var/lib
      --mandir=/usr/share/man
      --infodir=/usr/share/info
      --enable-fuse
      --enable-pixman
      --enable-painter
      --enable-vsock
      --enable-ipv6
      --with-socketdir=/run/xrdp
      --with-imlib2
      build_alias=x86_64-redhat-linux-gnu
      host_alias=x86_64-redhat-linux-gnu
      CC=gcc
      CFLAGS=-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64-v2 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
      LDFLAGS=-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1
      LT_SYS_LIBRARY_PATH=/usr/lib64:
      PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig

  Compiled with OpenSSL 3.0.7 1 Nov 2022
~~~


### Operating system & version

Red Hat Enterprise Linux release 9.3 (Plow)

### Installation method

dnf / apt / zypper / pkg / etc

### Which backend do you use?

tigervnc-server-minimal-1.13.1-2.el9.x86_64

### What desktop environment do you use?

GNOME 40 (x11)

### Environment xrdp running on

Physical PC with a Nvidia card 

### What's your client?

RDP client on Windows 11

### Area(s) with issue?

File transfer / drive redirection

### Steps to reproduce

On a RHEL 9.3 system, install xrdp from EPEL repo. 

In /etc/xrdp/sesman.ini, set:
```
[Chansrv]
FuseMountName=/media/thinclient_drives/%u/thinclient_drives
```
Then created the directory:
```
$ ls -l /media/
total 0
drwxrwxrwt. 2 root root 6 May  8 12:21 thinclient_drives
```


### ✔️ Expected Behavior

Selected Windows local drives should mount under the /media/thinclient_drives/%u/thinclient_drives/

### ❌ Actual Behavior

No directories created under the /media/thinclient_drives after the RDP session connected. Copy-paste works fine.

Also I see the xrdp-chansrv.10.log:
```
[20240508-12:51:46] [INFO ] Socket 12: AF_UNIX connection received
[20240508-12:51:46] [ERROR] Unable to obtain characteristics of directory containing /media/thinclient_drives/1947/thinclient_drives
[20240508-12:51:46] [ERROR] Unable to obtain characteristics of directory containing /media/thinclient_drives/1947/thinclient_drives
[20240508-12:51:46] [ERROR] Unable to obtain characteristics of directory containing /media/thinclient_drives/1947/thinclient_drives
[20240508-12:51:47] [INFO ] sound_process_output_format:
[20240508-12:51:47] [INFO ]       wFormatNo       0
[20240508-12:51:47] [INFO ]       wFormatTag      WAVE_FORMAT_PCM
[20240508-12:51:47] [INFO ]       nChannels       2
[20240508-12:51:47] [INFO ]       nSamplesPerSec  44100
[20240508-12:51:47] [INFO ]       nAvgBytesPerSec 176400
[20240508-12:51:47] [INFO ]       nBlockAlign     4
[20240508-12:51:47] [INFO ]       wBitsPerSample  16
[20240508-12:51:47] [INFO ]       cbSize          0
[20240508-12:51:47] [INFO ] sound_process_output_format:
[20240508-12:51:47] [INFO ]       wFormatNo       1
[20240508-12:51:47] [INFO ]       wFormatTag      WAVE_FORMAT_PCM
[20240508-12:51:47] [INFO ]       nChannels       2
[20240508-12:51:47] [INFO ]       nSamplesPerSec  22050
[20240508-12:51:47] [INFO ]       nAvgBytesPerSec 88200
[20240508-12:51:47] [INFO ]       nBlockAlign     4
[20240508-12:51:47] [INFO ]       wBitsPerSample  16
[20240508-12:51:47] [INFO ]       cbSize          0
[20240508-12:51:47] [INFO ] sound_process_training: round trip time 11
[20240508-12:51:47] [INFO ] Detected remote smartcard 'SCARD'
[20240508-12:51:47] [INFO ] Detected remote printer 'PRN4' (not supported)
[20240508-12:51:47] [INFO ] Detected remote printer 'PRN6' (not supported)
[20240508-12:51:47] [INFO ] Detected remote printer 'PRN5' (not supported)
[20240508-12:51:47] [INFO ] Detected remote drive 'N:'
[20240508-12:51:47] [INFO ] Detected remote drive 'E:'
```

### Anything else?

_No response_

--- a/sesman/chansrv/chansrv_fuse.c
+++ b/sesman/chansrv/chansrv_fuse.c
@@ -540,10 +540,25 @@ xfuse_init(void)
         return -1;
     }
 
-    g_snprintf(g_fuse_clipboard_path, 255, "%s/.clipboard", g_fuse_root_path);
+    g_snprintf(g_fuse_clipboard_path, sizeof(g_fuse_clipboard_path),
+               "%s/.clipboard", g_fuse_root_path);
+
+    /* if FUSE mount point does not exist, create it */
+    if (!g_directory_exist(g_fuse_root_path))
+    {
+        (void)g_create_path(g_fuse_root_path);
+        if (!g_create_dir(g_fuse_root_path))
+        {
+            LOG(LOG_LEVEL_ERROR, "mkdir %s failed (%s)",
+                g_fuse_root_path, g_get_strerror());
+            return -1;
+        }
+    }
 
     /* Get the characteristics of the parent directory of the FUSE mount
      * point. Used by xfuse_path_in_xfuse_fs() */
+    g_fuse_root_parent_dev = -1;
+    g_fuse_root_parent_ino = -1;
     p = (char *)g_strrchr(g_fuse_root_path, '/');
     if (p != NULL)
     {
@@ -554,11 +569,6 @@ xfuse_init(void)
         g_fuse_root_parent_ino = g_file_get_inode_num(g_fuse_root_path);
         *p = '/';
     }
-    else
-    {
-        g_fuse_root_parent_dev = -1;
-        g_fuse_root_parent_ino = -1;
-    }
 
     if (g_fuse_root_parent_dev == -1 || g_fuse_root_parent_ino == -1)
     {
@@ -566,18 +576,6 @@ xfuse_init(void)
             "Unable to obtain characteristics of directory containing %s",
             g_fuse_root_path);
         return -1;
-    }
-
-    /* if FUSE mount point does not exist, create it */
-    if (!g_directory_exist(g_fuse_root_path))
-    {
-        (void)g_create_path(g_fuse_root_path);
-        if (!g_create_dir(g_fuse_root_path))
-        {
-            LOG(LOG_LEVEL_ERROR, "mkdir %s failed. If %s is already mounted, you must "
-                "first unmount it", g_fuse_root_path, g_fuse_root_path);
-            return -1;
-        }
     }
 
     /* setup xrdp file system */
@@ -906,14 +904,16 @@ static int xfuse_init_lib(struct fuse_args *args)
 {
     if (fuse_parse_cmdline(args, &g_mount_point, 0, 0) < 0)
     {
-        LOG_DEVEL(LOG_LEVEL_ERROR, "fuse_parse_cmdline() failed");
+        LOG(LOG_LEVEL_ERROR, "fuse_parse_cmdline() failed");
         fuse_opt_free_args(args);
         return -1;
     }
 
     if ((g_ch = fuse_mount(g_mount_point, args)) == 0)
     {
-        LOG_DEVEL(LOG_LEVEL_ERROR, "fuse_mount() failed");
+        LOG(LOG_LEVEL_ERROR, "FUSE mount on %s failed."
+            " If %s is already mounted, you must first unmount it",
+            g_mount_point, g_mount_point);
         fuse_opt_free_args(args);
         return -1;
     }
@@ -921,7 +921,7 @@ static int xfuse_init_lib(struct fuse_args *args)
     g_se = fuse_lowlevel_new(args, &g_xfuse_ops, sizeof(g_xfuse_ops), 0);
     if (g_se == 0)
     {
-        LOG_DEVEL(LOG_LEVEL_ERROR, "fuse_lowlevel_new() failed");
+        LOG(LOG_LEVEL_ERROR, "fuse_lowlevel_new() failed");
         fuse_unmount(g_mount_point, g_ch);
         g_ch = 0;
         fuse_opt_free_args(args);

